### DISCUSSION TITLE
control pulseaudio volume with mpd?

### DISCUSSION THREAD

**Discussion by seancmalloy** [MusicPlayerDaemon/MPD#1857] [General]
is it possible to control pulseaudio volume with mpd? when i enable volume control in mpd (mixer_type software in alsa) i have to set pulseaudio volume to 100% (or actual volume too low) but its seems to degrade quality of audio. is there an alternative to using software mixer? whats the best way to control volume (and EQ)?

environment:

raspi4 - debian 11
mpd - 0.22.6
        outputs - alsa, https, icecast, and pulse raop (up2stream/airplay default sink)
pulseaudio - 16.99.1
mympd - 11.0.2-1

**Comment by MaxKellermann** [comment]
Yes, there's the PulseAudio mixer plugin that acts as a "hardware" mixer.

**Comment by seancmalloy** [comment]
thanks for response

when i try to add hardware mixer_type it shows volume control is disabled.

audio_output {
    type            "pulse"
    name            "Up2stream"
    sink            "raop_output.SoundSystem_20A5-2.local"
    format          "44100:16:2"
    mixer_type      "hardware"
}

![Screenshot (13)](https://github.com/MusicPlayerDaemon/MPD/assets/23019226/bd3c7aa7-c292-4b60-a64f-c54292c38391)

is this the plugin that you are talking about?

**Reply by MaxKellermann** [reply]
I don't know what this message means, I have never seen this client. What I see, however, is that you have an ancient unsupported MPD version.
Check https://www.musicpd.org/help/

**Comment by seancmalloy** [comment]
its mympd. ive been going back and forth between git build v0.23.13-792-gba2df05fb and mpd 0.22.6 (bullseye).  either way mixer_type hardware in pulse output doesnt control pulse volume, which is the goal. 

but i assume that this is not possible, and the mpd volume plugin just adjust volume based on pulseaudio volume, ie - if pulse is set at 50% and mpd is set to 50% its actual 25% volume 

am i wrong?

**Comment by jcorporation** [comment]
myMPD reports "Volumecontrol disabled", if the `status` or `getvol` command reports no volume. This usually appears if there is no valid mixer configured.
